src/msg/async/AsyncMessenger.cc

```diff
@@ -675,11 +682,15 @@ ConnectionRef AsyncMessenger::connect_to(int type, const entity_addrvec_t& addrs
 
   auto av = _filter_addrs(addrs);
 
+  if (anon) {
+    return create_connect(av, type, anon);
```

Comment:
Why don't we need to also check if an anonymous connection to the same peer already exists, as we do with non-anonymous connections?